library/Fission/User/Query.hs

```diff
@@ -0,0 +1,23 @@
+module Fission.User.Query
+  ( getHerkouAddonByUUID
+  , getHerkouAddonByUserId
+  ) where
+
+import           Data.UUID (UUID)
+import           Database.Esqueleto hiding ((=.), update)
+
+import           Fission.Models
+import           Fission.Prelude
+
+getHerkouAddonByUserId :: MonadDB m => HerokuAddOnId -> Transaction m [Entity User]
+getHerkouAddonByUserId addOnId = select <| from \user -> do
+      where_ <| user ^. UserHerokuAddOnId ==. val (Just addOnId)
+            &&. user ^. UserActive        ==. val True
+      limit 1
+      return user
+
+getHerkouAddonByUUID :: MonadDB m => UUID -> Transaction m [Entity HerokuAddOn]
+getHerkouAddonByUUID uuid = select <| from \herokuAddOn -> do
+      where_ (herokuAddOn ^. HerokuAddOnUuid ==. val uuid)
+      limit 1
```

Comment:
There's also a `selectFirst` function: https://www.stackage.org/haddock/lts-14.20/persistent-2.9.2/Database-Persist-Class.html#v:selectFirst

```haskell
selectFirst :: (MonadIO m, PersistRecordBackend record backend) => [Filter record] -> [SelectOpt record] -> Transaction m (Maybe (Entity record)) 
```

It's a nice fit because:
1. It returns a `Maybe`, which is more likely what you want rather than a `[]` that you later have to inspect)
2. Has the `limit` built-in